src/random_unitary.jl

```diff
@@ -0,0 +1,52 @@
+using MatrixAlgebraKit: qr_full!
+using Random: Random, AbstractRNG, randn!
+
+function square_zero_map(elt::Type, ax::Tuple{AbstractUnitRange,Vararg{AbstractUnitRange}})
+  return zeros(elt, (ax..., ax...))
+end
+
+function random_unitary!(rng::AbstractRNG, a::AbstractArray)
+  @assert iseven(ndims(a))
+  ndims_codomain = ndims(a) ÷ 2
+  biperm = blockedperm(ntuple(identity, ndims(a)), (ndims_codomain, ndims_codomain))
+  a_mat = fusedims(a, biperm)
+  random_unitary!(rng, a_mat)
+  splitdims!(a, a_mat, biperm)
+  return a
+end
+
+function random_unitary!(rng::AbstractRNG, a::AbstractMatrix)
```

Comment:
Is it a common enough pattern in Julia for a modifying (!) function to modify a different argument that its first one? I.e. should the arguments here be switched?

Reply:
Some arguments like the RNG take precedence over that convention, see for example: [Random.randn!](https://docs.julialang.org/en/v1/stdlib/Random/#Random.randn!).

Reply:
Good to know, thanks